### PR TITLE
Support Managed Identity in Azure Cache for Redis

### LINKED ISSUE #5794: Azure Redis should support managed identity
Similar to [Azure PostgreSQL should support managed identity (dotnet/aspire#5793)](https://github.com/dotnet/aspire/issues/5793), we should support managed identity when publishing an Azure Cache for Redis resource.

### Proposal

Add a new enum that gets passed into `.[Publish]AsAzureRedis()` that switches which kind of auth is used.

```C#
namespace Aspire.Hosting.Azure;

public enum AzureRedisAuth
{
    Password = 0,
    ActiveDirectory = 1 // Note the name used in bicep and ARM is `aad-enabled`
}

// user code
builder.AddRedis("cache")
    .PublishAsAzureRedis(AzureRedisAuth.ActiveDirectory);
```

cc @mitchdenny @davidfowl @tg-msft @sebastienros @radical

### PR DESCRIPTION
**Pull request by eerhardt**
## Description

Similar to https://github.com/dotnet/aspire/pull/5930, we add a new API pattern for working with Azure resources that may also be local containers.

- Add a new API to replace AsAzure and PublishAsAzure. This creates an Azure resource to start, so it can be configured like other azure resources. Since this is a new API, we can use managed identity by default.

- Add an API to convert managed identity to password based auth, so users can still go that route if they need to.

- Add RunAsContainer following the RunAsEmulator model. This allows local development against a Redis container and publish to go to a manged Azure Cache for Redis.

Fix #5794

## Checklist

- Is this feature complete?
  - [x] No. Follow-up changes expected. Need to update PostgreSQL and SqlServer to follow this new pattern.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [x] Yes
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [x] Yes
- Does the change make any security assumptions or guarantees?
  - [x] Yes
    - If yes, have you done a threat model and had a security review?
      - [x] Yes
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Link to aspire-docs issue: https://github.com/dotnet/docs-aspire/issues/1704

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5966)

### CODE REVIEW

File: src/Aspire.Hosting.Azure.Redis/Aspire.Hosting.Azure.Redis.csproj [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <PackageTags>aspire integration hosting azure</PackageTags>
     <Description>Azure Redis resource types for .NET Aspire.</Description>
     <PackageIconFullPath>$(SharedDir)AzureCacheRedis_256x.png</PackageIconFullPath>
+    <UsePublicApiAnalyzers>false</UsePublicApiAnalyzers>
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.Redis/Aspire.Hosting.Azure.Redis.csproj:9]: 😄

Comment (eerhardt) [reply src/Aspire.Hosting.Azure.Redis/Aspire.Hosting.Azure.Redis.csproj:9]: Gotta get work done somehow....

**Comment by eerhardt** [comment]
This PR is ready for review. I believe I have everything ready for this to be merged.

If you want to see this in action - https://github.com/eerhardt/aspire/compare/RedisManagedIdentity...eerhardt:aspire:UseManagedIdentityInPlayground is what it would look like to update the playground app to use Managed Identity in Azure, and a local Redis container for run.

File: src/Aspire.Hosting.Azure.Redis/AzureRedisResource.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ namespace Aspire.Hosting.Azure;
 /// </summary>
 /// <param name="innerResource">The inner resource.</param>
 /// <param name="configureConstruct">Callback to populate the construct with Azure resources.</param>
+[Obsolete($"This method is obsolete and will be removed in a future version. Use {nameof(AzureRedisExtensions.AddAzureRedis)} instead to add an Azure Cache for Redis resource.")]
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.Redis/AzureRedisResource.cs:13]:
```suggestion
[Obsolete($"This class is obsolete and will be removed in a future version. Use {nameof(AzureRedisExtensions.AddAzureRedis)} instead to add an Azure Cache for Redis resource.")]
```

**Review by davidfowl — APPROVED** [review]

**Pull request merged**